libraries/wbInterface/src/SetPIDBlock.cpp

```diff
@@ -0,0 +1,136 @@
+#include "SetPIDBlock.h"
+
+#include <yarp/os/ResourceFinder.h>
+#include <codyco/PIDList.h>
+#include <yarpWholeBodyInterface/yarpWholeBodyInterface.h>
+#include <sstream>
+
+namespace wbitoolbox {
+    using namespace codyco;
+
+    const std::string TorquePIDInitialKey = "__ORIGINAL_PIDs__";
+    const std::string TorquePIDDefaultKey = "__DEFAULT_PIDs__";
+
+    static bool loadTorqueGainsFromFile(std::string filename,
+                                        const PIDList &originalList,
+                                        yarpWbi::yarpWholeBodyInterface& interface,
+                                        PIDList &loadedPIDs);
+    static bool fillPIDWithBottleDescription(const yarp::os::Bottle &bottle, yarp::dev::Pid &pid);
+
+
+    bool loadGainsFromValue(const yarp::os::Value &gains,
+                               PidMap &pidMap,
+                               yarpWbi::yarpWholeBodyInterface& interface)
+    {
+        pidMap.clear();
+
+        //Load original gains from controlboards and save them the original key.
+        PIDList originalGains(interface.getDoFs());
+        yarpWbi::yarpWholeBodyActuators *actuators = interface.wholeBodyActuator();
+        actuators->getPIDGains(originalGains.pidList(), wbi::CTRL_MODE_TORQUE);
+        pidMap.insert(PidMap::value_type(TorquePIDInitialKey, originalGains));
+
+        //Now load additional gains
+        bool result = true;
+        if (gains.isString()) {
+            PIDList pids(interface.getDoFs());
+            result = loadTorqueGainsFromFile(gains.asString(), originalGains, interface, pids);
+            pidMap.insert(PidMap::value_type(TorquePIDDefaultKey, pids));
+        } else if (gains.isList()) {
+            using namespace yarp::os;
+            Bottle *list = gains.asList();
+
+            //list of files. gains will be saved as integer-values
+            for (int i = 0; i < list->size(); ++i) {
+                if (!list->get(i).isString()) continue;
+                std::string filename = list->get(i).asString();
+                PIDList pids(interface.getDoFs());
+                result = loadTorqueGainsFromFile(filename, originalGains, interface, pids);
+
+                if (result) {
+                    std::ostringstream num;
+                    num << (i + 1);
+                    pidMap.insert(PidMap::value_type(num.str(), pids));
+                }
+            }
+        }
+        return result;
+    }
+
+
+    bool setCurrentGains(const PidMap &pidMap, int map, yarpWbi::yarpWholeBodyInterface& interface)
+    {
+        std::stringstream num;
```

Comment:
@francesco-romano could it be a good idea to incapslate this string2int conversion in a function? It would make it easier to migrate to `std::to_string` when we will go (if ever) C++11 only.